### DISCUSSION TITLE
JsonWebToken and thread safety with kotlin co-routines

### DISCUSSION THREAD

**Discussion by dodalovicgran** [quarkusio/quarkus#38699] [Q&A] [answered]
HI, I'm using Quarkus, with Kotlin and suspending functions. 

Say I have something like:

```kotlin
@Path("/temp")
class TempResource(private val jwt: JsonWebToken) {

    private val log = KotlinLogging.logger {}

    @GET
    suspend fun temp(): String {
        log.info { jwt.subject }
        return "temp"
    }
}
```

Do I have to do anything on top to make sure that jwt.subject will always be resolved to the subject of the currently processed request? 

I saw some thing called `quarkus-smallrye-context-propagation`, but not sure if I need it or not. 

Thanks in advance :bow:

[a comment by an automated account is not shown]

**Comment by geoand** [comment, marked as answer]
It should just work without having to do anything

**Reply by dodalovicgran** [reply]
Thank You Georgios!

**Reply by geoand** [reply]
🙏🏼
